### ISSUE TITLE
Benchmark against unsloth

### ISSUE THREAD

**Issue by ccdv-ai** [linkedin/Liger-Kernel#57]
### 🚀 The feature, motivation and pitch

hey, did you run any benchmark against unsloth which uses similar kernels?
I guess your project can be used as a dropdown replacement with multi gpu support.

**Comment by yundai424** [comment]
Hi @ccdv-ai thanks for your raising this issue! We had previously done internal benchmark against various open source triton kernel implementations available (including some unsloth kernels) to make sure the performance is not behind the community :) We actually get quite some inspirations from unsloth tbh!

There are certainly some differences between unsloth and liger though. 1) Unsloth works well on single GPU and currently has a wider coverage. We have not looked into LoRA yet which unsloth does a great job on. Right now we're targeting more on multi-GPU full-param training but LoRA and MoE are certainly interesting topics that we want to explore as well. 2) Also Unsloth is like a one stop shop to do everything for you but liger is more like some drop-in kernel replacement and users still need to figure out what trainer / training loop etc. to use.

Anyway, please find the benchmark results below (these were performed on one A100 GPU). Also we have our benchmark suite in [`/benchmark`] directory ([example](https://github.com/linkedin/Liger-Kernel/blob/main/benchmark/benchmark_rope.py)) and it's very easy to plug in other kernels into the benchmark function too by defining another impl [provider](https://github.com/linkedin/Liger-Kernel/blob/main/benchmark/benchmark_rope.py#L83), please feel free to try it yourself! We would love to see how the performance look like on various hardware to help us further improve the implementation too :)

cross entropy full (fwd+bwd) speed:
![image](https://github.com/user-attachments/assets/3d11cde4-2f8d-4c95-84b3-5295531577d5)

cross entropy full (fwd+bwd) GPU RAM usage:
![image](https://github.com/user-attachments/assets/816d50e0-d1a3-4931-9647-55ac8e7b19cb)

RoPE full (fwd+bwd) speed, hidden dim set as 8192:
![image](https://github.com/user-attachments/assets/21f0e4f5-8c35-4e82-90d3-de36187d4f96)

RoPE full (fwd+bwd) GPU RAM, hidden dim set as 8192:
![image](https://github.com/user-attachments/assets/28158830-6f20-4b16-b486-5253c25a3ae9)

(liger and unsloth overlap on both GPU memory benchmark)

**Comment by ByronHsu** [comment]
Closed the ticket now. Thanks!
